nab/detectors/htmcore/setup.py

```diff
@@ -0,0 +1,115 @@
+# ----------------------------------------------------------------------
+# Copyright (C) 2014-2015, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import os
+import sys
+import pkg_resources
+import warnings
+from setuptools import setup, find_packages
+
+REPO_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+# Utility function to read the README file.
+# Used for the long_description.  It"s nice, because now 1) we have a top level
+# README file and 2) it"s easier to type in the README file than to put a raw
+# string in below ...
+def read(fname):
+  with open(os.path.join(os.path.dirname(__file__), fname)) as f:
+    result = f.read()
+  return result
+
+
+
+def nupicInstalled():
+  """
+  Determine whether NuPIC is already installed.
+   :return: boolean
+  """
+  try:
+    _ = pkg_resources.get_distribution("htm.core")
+    return True
+  except pkg_resources.DistributionNotFound:
+    return False  # Silently ignore. NuPIC will be installed later.
+
+
+def parseFile(requirementFile):
+  """
+  Parse requirement file.
+  :return: list of requirements.
+  """
+  try:
+    return [
+      line.strip()
+      for line in open(requirementFile).readlines()
+      if not line.startswith("#")
+    ]
+  except IOError:
+    return []
+
+
+
+def findRequirements():
+  """
+  Read the requirements.txt file and parse into requirements for setup's
+  install_requirements option.
+  """
+  requirements = parseFile(os.path.join(REPO_DIR, "requirements.txt")) #REPO_DIR is local htmcore/ "repo"
+
+  if not nupicInstalled():
+    # The user already has a version of htm.core (NuPIC) installed. If not, we'll attempt custom install from git.
+    # custom install htm.core #TODO when htm.core is published on PyPI, remove this whole custom setup.py, and just add htm.core to REPO/requirements.txt
+    try:
+      os.system("git status")
+    except:
+      raise "Git must be installed for htm.core detector installation!"
+
+    curr_dir = os.getcwd()
+    os.chdir(REPO_DIR)
+    os.system("git clone --depth=5 https://github.com/htm-community/htm.core")
```

Comment:
when htm.core can be available on PyPI, skip all this mess https://github.com/htm-community/htm.core/issues/19